wandb/sdk/service/service_grpc.py

```diff
@@ -0,0 +1,69 @@
+"""grpc service.
+
+Reliably launch and connect to grpc process.
+"""
+
+import logging
+from typing import Optional
+from typing import TYPE_CHECKING
+
+import grpc
+from wandb.proto import wandb_server_pb2 as spb
+from wandb.proto import wandb_server_pb2_grpc as pbgrpc
+
+from .service_base import _pbmap_apply_dict
+from .service_base import ServiceInterface
+
+if TYPE_CHECKING:
+    from wandb.sdk.wandb_settings import Settings
+
+
+class ServiceGrpcInterface(ServiceInterface):
+    _stub: Optional[pbgrpc.InternalServiceStub]
+
+    def __init__(self) -> None:
+        self._stub = None
+
+    def get_transport(self) -> str:
+        return "grpc"
+
+    def _svc_connect(self, port: int) -> None:
+        channel = grpc.insecure_channel("localhost:{}".format(port))
+        stub = pbgrpc.InternalServiceStub(channel)
+        self._stub = stub
+        # TODO: make sure service is up
```

Comment:
should we address this TODO?

Reply:
Not now since GRPC isnt the primary.   There are many things that need to be done for grpc... like handing timeouts and retries.  This would require making all the handler message handling idempotent or some higher level handling of retries.